### PR TITLE
Add Nano32 development board

### PR DESCRIPTION
**Pull request by NAzT**

### CODE REVIEW

**Comment by me-no-dev** [comment]
Since the configuration is exactly the same as the currently added module, do you mind holding a bit for us to add a bit more features and config options? It will be much easier not to miss some option on a device if we have just one :)

**Comment by ESP32DE** [comment]
you are right the configuration is the same
the Nano32 development board pin description is as Pull Request attached
The difference are 

differ_part1

differ_part2_git

screen-shot

File: boards.txt [hunk 1]
```diff
@@ -40,3 +40,44 @@ esp32.menu.UploadSpeed.460800.macosx=460800
...
 esp32.menu.UploadSpeed.512000.windows=512000
 esp32.menu.UploadSpeed.512000.upload.speed=512000
+
+
+##############################################################
+nano32.name=Nano32
+
+nano32.upload.tool=esptool
+nano32.upload.maximum_size=1044464
+nano32.upload.maximum_data_size=114688
```
Comment (me-no-dev) [inline boards.txt]: please change upload.maximum_data_size to 294912

Comment (NAzT) [inline boards.txt]: Changed.

Thank you 👍

**Comment by me-no-dev** [comment]
check you PR :) 33 commits 27 files! Close this and try again maybe clean

**Pull request closed without merging**